project/src/main/java/com/google/sps/data/ServletLibrary.java

```diff
@@ -42,18 +65,45 @@ public static Account retrieveAccountInfo(DatastoreService datastore, UserServic
 
     if (entity == null) return null;
 
-    String logoutUrl = userService.createLogoutURL("/index.html");
-    String nickname = entity.getProperty("nickname").toString();
-    String userEmail = entity.getProperty("userEmail").toString();
-    boolean isUserBusinessOwner = (boolean) entity.getProperty("isUserBusinessOwner");
+    // Retrieve entity properties, and check if they are valid types.
+    Object _nickname = entity.getProperty("nickname");
+    Object _userEmail = entity.getProperty("userEmail");
+    Object _isUserBusinessOwner = entity.getProperty("isUserBusinessOwner");
+    Object _street = entity.getProperty("street");
+    Object _city = entity.getProperty("city");
+    Object _state = entity.getProperty("state");
+    Object _zipCode = entity.getProperty("zipCode");
+    String nickname;
+    String userEmail;
+    boolean isUserBusinessOwner;
+    String street;
+    String city;
+    String state;
+    String zipCode;
+
+    if ((_nickname instanceof String) && 
+        (_userEmail instanceof String) && 
+        (_isUserBusinessOwner instanceof Boolean) &&
+        (_street instanceof String) &&
+        (_city instanceof String) &&
+        (_state instanceof String) &&
+        (_zipCode instanceof String)) {
+      nickname = _nickname.toString();
+      userEmail = _userEmail.toString();
+      isUserBusinessOwner = (boolean) _isUserBusinessOwner;
+      street = _street.toString();
+      city = _city.toString();
+      state = _state.toString();
+      zipCode = _zipCode.toString();
+    } else {
+      System.err.println("Entity properties are of an incorrect type.");
```

Comment:
This would be easier to debug in the future if you broke it down so you could log the specific entity that was the wrong type. I'm fine with it this way though, just wanted to point it out for a possible improvement.

Reply:
True I could do that! I chose not to do that for now as it would require having like 30 different if statements, but it is definitely good to have it that way for debugging quicker.